calblueprint/ogsc
# Apply filterPlayerProfileRead/write to API endpoints

* Builds off of work from #93 to use `filterPlayerProfileRead` and `filterPlayerProfileWrite` functions in relevant API endpoints: 
  * Search users endpoint
  * Update users endpoint
* Moves search endpoint out of admin/ API namespace (search is used by multiple user roles)
* Updates table in PlayersDashboard to use player number profile field rather than numeric ID
* (chore) Bumps caniuse-db in Yarn lockfile

## TODO

* To be resolved in separate PR: editing UI states must be updated to reflect lib/access definitions defined on a per-user role basis
* To be resolved in separate PR: many of the API endpoints related to updating profile fields are not admin-only and should be moved out of the /api/admin namespace to reflect that

## How to Test/Use PR feature

* Test that API responses do not include fields that should not be accessible to the current user type, e.g.:
  *  Open the Chrome DevTools > Network tab. Keep this open.
  * Login in as a player.
  * Visit the players index page.
  * Search for a player.
  * Click on the /api/players/search request in the Network tab, and view the Response. The response JSON _should not_ include profile field details that players do not have access for each other: health and wellness details, for example.

## PR Checklist

Does your branch (check if true):

## Related PRs

* #93 for more context on lib/access or filterPlayerProfileRead/Write.



CC: @sydbui
